pkg/configtx/msp.go

```diff
@@ -607,6 +633,54 @@ func pemEncodePKCS8PrivateKey(priv crypto.PrivateKey) ([]byte, error) {
 	return pem.EncodeToMemory(&pem.Block{Type: "PRIVATE KEY", Bytes: privBytes}), nil
 }
 
+// UpdateConsortiumMSP updates the MSP config for the specified consortium org group.
+func (c *ConfigTx) UpdateConsortiumMSP(updatedMSP MSP, consortiumName string, orgName string) error {
+	currentMSP, err := c.ConsortiumMSP(consortiumName, orgName)
+	if err != nil {
+		return fmt.Errorf("retrieving msp: %v", err)
+	}
+
+	if currentMSP.Name != updatedMSP.Name {
+		return errors.New("MSP name cannot be changed")
+	}
+
+	err = updatedMSP.validateCACerts()
+	if err != nil {
+		return err
+	}
+
+	err = setMSPConfigForConsortium(c.UpdatedConfig(), updatedMSP, consortiumName, orgName)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// UpdateOrdererMSP updates the MSP config for the specified orderer org group.
+func (c *ConfigTx) UpdateOrdererMSP(updatedMSP MSP, orgName string) error {
+	currentMSP, err := c.OrdererMSP(orgName)
+	if err != nil {
+		return fmt.Errorf("retrieving msp: %v", err)
+	}
+
+	if currentMSP.Name != updatedMSP.Name {
+		return errors.New("MSP name cannot be changed")
+	}
+
+	err = updatedMSP.validateCACerts()
+	if err != nil {
+		return err
+	}
+
+	err = setMSPConfigForOrderer(c.UpdatedConfig(), updatedMSP, orgName)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
 // UpdateApplicationMSP updates the MSP config for the specified application
 // org group.
 func (c *ConfigTx) UpdateApplicationMSP(updatedMSP MSP, orgName string) error {
```

Comment:
Not part of your commit but might as well update this to `SetApplicationMSP` as well